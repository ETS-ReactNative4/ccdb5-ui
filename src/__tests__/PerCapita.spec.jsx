import configureMockStore from 'redux-mock-store'
import {
  mapDispatchToProps, mapStateToProps, PerCapita
} from '../PerCapita'
import { Provider } from 'react-redux'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import thunk from 'redux-thunk'

function setupSnapshot() {
  const middlewares = [ thunk ]
  const mockStore = configureMockStore( middlewares )
  const store = mockStore( {
    map: {}
  } )

  return renderer.create(
    <Provider store={ store }>
      <PerCapita />
    </Provider>
  )
}

describe( 'component: PerCapita', () => {
  describe( 'initial state', () => {
    it( 'renders without crashing', () => {
      const target = setupSnapshot()
      let tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )
  } )

  describe('mapDispatchToProps', () => {
    it('hooks into onDataNormalization', () => {
      const dispatch = jest.fn();
      const ev = {
        target: {
          value: 123
        }
      }
      mapDispatchToProps(dispatch).onDataNormalization( ev );
      expect(dispatch.mock.calls.length).toEqual(1);
    })
  })

  describe( 'mapStateToProps', () => {
    it( 'maps state and props', () => {
      const state = {
        map: {
          dataNormalization: 'foo'
        },
        query: {}
      }
      let actual = mapStateToProps( state )
      expect( actual ).toEqual( {
        dataNormalization: 'foo',
        enablePer1000: true
      } )
    } )
  } )


} )
