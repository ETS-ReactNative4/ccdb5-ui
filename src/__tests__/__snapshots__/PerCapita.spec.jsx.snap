// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: PerCapita initial state renders without crashing 1`] = `
<section
  className="per-capita"
>
  <div
    className="per-capita m-btn-group"
  >
    <p>
      Map Shading
    </p>
    <label
      className="u-visually-hidden"
      htmlFor="data-normalization"
    >
      Select the kind of shading to apply to the map
    </label>
    <button
      className="a-btn raw deselected"
      onClick={[Function]}
    >
      Complaints
    </button>
    <button
      className="a-btn capita a-btn__disabled"
      onClick={[Function]}
    >
      Complaints per 1,000 population
    </button>
  </div>
</section>
`;
